mock/src/app.ts

```diff
@@ -0,0 +1,243 @@
+    list_endpoint: '/centreon/api/latest/data-source/metaservices',
+  },
+  { slug: 'metric', display_name: 'metric', list_endpoint: '/centreon/api/latest/data-source/metrics' },
+  { slug: 'service', display_name: 'service', list_endpoint: '/centreon/api/latest/data-source/services' },
+  {
+    slug: 'service-group',
+    display_name: 'service group',
+    list_endpoint: '/centreon/api/latest/data-source/service-groups',
+  },
+  {
+    slug: 'virtual-metric',
+    display_name: 'virtual metric',
+    list_endpoint: '/centreon/api/latest/data-source/virtual-metrics',
+  },
+];
+
+dataSourceRouter.get('/types', ensureAuthenticated, (req, res) => {
+  res.send(resourcesTypes);
+});
+
+const resourcesTypesExamples: Record<string, Array<{ name: string; [key: string]: string }>> = {
+  host: [
+    {
+      name: 'Centreon-central',
+    },
+  ],
+  service: [
+    {
+      name: 'Ping',
+    },
+    {
+      name: 'Cpu',
+    },
+    {
+      name: 'Memory',
+    },
+  ],
+  ['host-group']: [
+    {
+      name: 'HG',
+    },
+  ],
+  ['service-group']: [
+    {
+      name: 'SG',
+    },
+  ],
+  ['virtual-metric']: [
+    {
+      name: 'VM',
+    },
+  ],
+  ['business-activity']: [
+    {
+      name: 'BA',
+    },
+  ],
+  ['anomaly-detection']: [
+    {
+      name: 'AD',
+    },
+  ],
+  metaservice: [
+    {
+      name: 'metaservice',
+    },
+  ],
+  metric: [
+    {
+      name: 'metric',
+    },
+  ],
+};
+
+const createEndpoint = ({ list_endpoint, slug }: MBIResourceType) => {
+  dataSourceRouter.get(`/${list_endpoint.split('/').pop()}`, ensureAuthenticated, (req: express.Request, res) => {
+    const name = req.query[slug]?.toString()?.replace('*', '') || '';
+
+    const resourcesTypesCorrespondingToRequest = resourcesTypesExamples[slug] || [];
+    const result = [...new Array(getRandomArbitrary(1, 15))].map(() => {
+      const id = getRandomArbitrary(1, 1500);
+      const randomResource =
+        resourcesTypesCorrespondingToRequest[Math.floor(Math.random() * resourcesTypesCorrespondingToRequest.length)];
+      return {
+        ...randomResource,
+        id,
+        name: `${name ? name + '-' : ''}${randomResource.name}-${id}`,
+      };
+    });
+
+    res.send({
+      result: result,
+      meta: {
+        page: 1,
+        limit: 10,
+        search: {},
+        sort_by: {},
+        total: result.length,
+      },
+    });
+  });
+};
+
+resourcesTypes.forEach((type) => {
+  createEndpoint(type);
+});
+
+const metricsSample = ['response_time', 'load5', 'load10', 'load15'];
+
+dataSourceRouter.get('/metrics/timeseries', ensureAuthenticated, (req, res) => {
+  const { metrics } = req.query;
+
+  const from = new Date(req.query.start?.toString() || Date.now() - 3 * 60 * 60 * 1000);
+  const to = new Date(req.query.end?.toString() || Date.now());
+
+  let returnMetrics = metrics
+    ? Array.isArray(metrics)
+      ? metrics.map((m) => m.toString())
+      : [metrics.toString()]
+    : metricsSample;
```

Comment:
@bdauria better than a mutation ? ( but it's a ternary in ternary ... )

Reply:
You can extract this into a function and implement the same logic in a cleaner way